pkg/releases/consumer_test.go

```diff
@@ -0,0 +1,161 @@
+package releases
+
+import (
+	"fmt"
+	"log"
+	"net/http"
+	"testing"
+
+	"github.com/atlassian/voyager/pkg/releases/deployinator/models"
+	"github.com/pact-foundation/pact-go/dsl"
+)
+
+/**
+This test tests the consumer interactions surrounding the resolve endpoints between
+voyager (consumer) and trebuchet (provider).
+
+To validate the pact against the current provider specs locally, use the contract testing cli found here:
+https://hello.atlassian.net/wiki/spaces/TESTA/pages/128609304/Contract+Testing+Getting+Started+Guide+for+Consumers
+ */
+func TestConsumerPact(t *testing.T) {
+
+	// Initialize the pact
+	pact := &dsl.Pact{
+		Consumer: "voyager",
+		Provider: "deployinator-trebuchet",
+		Host:     "localhost",
+		PactDir:  "generated/pacts",
+	}
+
+	defer pact.Teardown()
+
+	// Define the interactions in resolve and batchResolve separately, with the same pact
+
+	ResolveEndpointTest(pact)
+
+	BatchResolveEndpointTest(pact)
+
+	// Once interactions are verified successfully,
+	// a pact contract between the voyager/trebuchet should have now been generated. See contract under pacts/
+}
+
+func ResolveEndpointTest(pact *dsl.Pact) {
+	singleReleaseGroup := models.ResolutionResponseType{
+		Label:         "",
+		Service:       "ServiceName",
+		ReleaseGroups: map[string]map[string]interface{}{},
+	}
+
+	// Define request with dummy variables and headers set
+	var resolveTest = func() error {
+		u := fmt.Sprintf("http://%s:%d/api/v1/resolve", pact.Host, pact.Server.Port)
+		req, err := http.NewRequest("GET", u, nil)
+		q := req.URL.Query()
+		q.Add("service", "serviceName")
+		q.Add("environment", "dev")
+		q.Add("region", "us-east-1")
+		q.Add("account", "1233")
+		req.URL.RawQuery = q.Encode()
+
+		if err != nil {
```

Comment:
Error checking should be right after the place where an error might have happened: `req, err := http.NewRequest("GET", u, nil)`